crates/biome_js_analyze/src/lint/nursery/use_import_extensions.rs

```diff
@@ -0,0 +1,201 @@
+            }),
+        )
+    }
+
+    fn action(ctx: &RuleContext<Self>, state: &Self::State) -> Option<JsRuleAction> {
+        let mut mutation = ctx.root().begin();
+        let node = ctx.query();
+
+        mutation.replace_node(
+            node.clone(),
+            make::js_module_source(make::js_string_literal(&state.suggestion)),
+        );
+
+        Some(JsRuleAction {
+            category: ActionCategory::QuickFix,
+            applicability: Applicability::MaybeIncorrect,
+            message: markup! {
+                "Add potential import extension "<Emphasis>"."{state.potential_import_ext}</Emphasis>"."
+            }.to_owned(),
+            mutation,
+        })
+    }
+}
+
+pub struct UseImportExtensionsState {
+    potential_import_ext: String,
+    suggestion: String,
+}
+
+const GUARANTEED_INDEX_FILE_ENDINGS: &[&str] = &["/", "..", "."];
+
+fn get_extensionless_import(
+    file_ext: &str,
+    module_path: &TokenText,
+) -> Option<UseImportExtensionsState> {
+    if !module_path.starts_with('.') {
+        return None;
+    }
+
+    if Path::new(module_path.text()).extension().is_some() {
+        return None;
+    }
+
+    let mut path_parts: Vec<_> = module_path.text().split('/').collect();
+
+    if module_path.ends_with('/') {
+        path_parts.pop();
+    }
+
+    if GUARANTEED_INDEX_FILE_ENDINGS
+        .iter()
+        .any(|ending| module_path.ends_with(ending))
+    {
+        // 'import ".././"' is the same as 'import "../"'
+        if path_parts.len() > 1 && path_parts.last() == Some(&".") {
+            path_parts.pop();
+        }
+
+        let import_ext = resolve_import_extension(file_ext, &path_parts);
+        let part = format!("index.{}", import_ext);
+        path_parts.push(part.as_str());
+
+        return Some(UseImportExtensionsState {
+            potential_import_ext: import_ext.to_owned(),
+            suggestion: path_parts.join("/"),
+        });
+    }
+
+    let import_ext = resolve_import_extension(file_ext, &path_parts);
+
+    let last = path_parts.last_mut()?;
+    let new_last = format!("{}.{}", last, import_ext);
+    *last = new_last.as_str();
+
+    Some(UseImportExtensionsState {
+        potential_import_ext: import_ext.to_owned(),
+        suggestion: path_parts.join("/"),
+    })
+}
+
+/// If import base name is uppercase, it is likely a component.
+fn is_potential_component(path_parts: &'_ [&str]) -> bool {
+    get_basename(path_parts)
+        .and_then(|base_name| base_name.chars().next())
+        .is_some_and(|c| c.is_uppercase())
+}
+
+fn resolve_import_extension<'a>(file_ext: &str, path_parts: &'_ [&'a str]) -> &'a str {
+    // TODO. This is not very accurate. We should use file system access to determine the file type.
+    let (potential_ext, potential_component_ext) = match file_ext {
+        "ts" | "tsx" | "astro" => ("ts", "tsx"),
+        "mts" => ("mts", "tsx"),
+        "mjs" => ("mjs", "jsx"),
```

Comment:
You should add `cjs` and `cts` too

Reply:
Can `cjs` and `cts` include `import` statements?

Reply:
You can with `require()`: `require("./file.js")`